### DISCUSSION TITLE
Request help on how to share the socket connection in multiple html pages

### DISCUSSION THREAD

**Discussion by bug-fixed** [miguelgrinberg/Flask-SocketIO#2043] [Q&A]
Hi there,

I'm trying to develop a simple website. The environment is Flask, Flask-socketio, jquery, html, javascript, Apache httpd server with wSGI daemon.  The jquery is used to transfer data (json format) between the server and the client html pages.

I'm stuck in the implementation of a progress bar for some time-consuming functions. Thanks to the flask-socketio and following https://www.youtube.com/watch?v=bUfUKtJqaxQ, I can figure out a simple progressbar, it runs OK in the html pages. 

For example, I have three html pages, `index.html`, `A.html` and `B.html`. Both `A.html` and `B.html` have a progressbar respectively. In each of them, one independent socket connection was created and listens the specified message emitted from the server. 

However, it seems the html pages will be stuck if both `A` and `B` are opened in one browser. Is it possible to create one single socket connection for one user/client? For example, in the `index.html`, one global socket connection was created and shared with `A.html` and `B.html`? I've tried searching in Google, and no fortunate got it worked. 

Could you please share some ideas for my case? Much appreciated!

**Comment by miguelgrinberg** [comment]
Read about rooms in the documentation. You cannot share a connection, but you can group your clients into rooms and emit messages to them together.

**Reply by bug-fixed** [reply]
Hi Mr. @miguelgrinberg , thanks a lot for your quick reply and kind suggestions! I'm really a newbie in the web development.
Finally, I've got it worked following your previously shared blog https://blog.miguelgrinberg.com/post/using-celery-with-flask and your shared GitHub repository. Your blog is really helpful! Now I'm using the celery framework for the task progress status update. Flask-socketio is really helpful! Currently I just need a runnable demo, maybe I will resort to flask-socketio if it is necessary in the future.
Thanks a lot!
